### PR TITLE
TES-309: User data script

### PR DESCRIPTION
**Pull request by mihailradkov**
Changes:

- Additional repo configurations
- Removed unused variables and resources
- TES-308: Organized the VM module
- TES-309: Initial user data script and load balancer module

### CODE REVIEW

File: modules/vm/main.tf [hunk 1]
```diff
@@ -101,10 +30,10 @@ resource "azurerm_network_security_group" "graphdb" {
     direction                  = "Inbound"
     access                     = "Allow"
     protocol                   = "Tcp"
-    source_port_range          = "7200"
-    destination_port_range     = "7201"
-    source_address_prefixes    = local.subnet_cidr_blocks
-    destination_address_prefix = "*"
+    source_port_range          = "7200-7201"
```
Comment (mihailradkov) [inline modules/vm/main.tf:33]: Azure recommends to use * for source ranges..

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,99 @@
...
+# ------------------------------------------------------------
+
+# TODO: Could go into another module
+
+# TODO: Config + how to .. refer other account group?
+data "azurerm_resource_group" "image" {
+  name = "Packer-RG"
+}
+
+# TODO: Support for multiple architectures
```
Comment (rdstn) [inline main.tf:54]: TODO as well - when we release the image officially, we can move to [platform_image](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/platform_image).

File: modules/vm/main.tf [hunk 3]
```diff
@@ -1,98 +1,27 @@
...
+  resource_group = data.azurerm_resource_group.graphdb.name
+  location       = data.azurerm_resource_group.graphdb.location
+
+  subnet_id   = data.azurerm_subnet.graphdb.id
+  subnet_cidr = data.azurerm_subnet.graphdb.address_prefix
 }
 
+# TODO: Move out of here to a sg module ?
 # Create Network Security Group and rules
 resource "azurerm_network_security_group" "graphdb" {
```
Comment (mihailradkov) [inline modules/vm/main.tf:21]: We need a rule to allow inbound traffic to the load balancer

Comment (mihailradkov) [reply modules/vm/main.tf:21]: Will do later

File: modules/vm/main.tf [hunk 4]
```diff
@@ -142,46 +71,67 @@ resource "azurerm_network_security_group" "graphdb" {
...
       public_ip_address {
         name = "first"
       }
     }
   }
+
   os_disk {
+    # TODO: size? caching?
     caching              = "None"
     storage_account_type = "Premium_LRS"
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply modules/vm/main.tf:128]: Will decide later

**Pull request merged**